_data-prepper/pipelines/configuration/sinks/opensearch.md

```diff
@@ -29,16 +67,219 @@ index | Conditionally | String | Name of the export index. Applicable and requir
+
+```yaml
+sink:
+  - opensearch:
+      username: "admin"
+      password: "admin"
+      ...
+```
+
+Alternately, rather than Admin credentials, you can specify the credentials of a user that's mapped to a role with the following minimum permissions:
+
+### Cluster permissions
+
+- `cluster_all`
+- `indices:admin/template/get`
+- `indices:admin/template/put`
+
+### Index permissions
+
+- Index: `otel-v1*`; Index permission: `indices_all`
+- Index: `.opendistro-ism-config`; Index permission: `indices_all`
+- Index: `*`; Index permission: `manage_aliases`
+
+For instructions to map users to roles, see [Map users to roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles/#map-users-to-roles).
+
+## Amazon OpenSearch Service domain security
+
+The OpenSearch sink plugin can send data to an [Amazon OpenSearch Service](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/what-is.html) domain, which uses Identity and Access Management (IAM) for security. The plugin uses the default credential chain. Run `aws configure` using the [AWS CLI](https://aws.amazon.com/cli/) to set your credentials.
+
+Make sure the credentials that you configure have the required IAM permissions. The following domain access policy demonstrates the minimum required permissions for the sink to work:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::<AccountId>:user/data-prepper-user"
+      },
+      "Action": "es:ESHttp*",
+      "Resource": [
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_template/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_plugins/_ism/policies/raw-span-policy",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_alias/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_alias/_bulk"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::<AccountId>:user/data-prepper-user"
+      },
+      "Action": "es:ESHttpGet",
+      "Resource": "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_cluster/settings"
+    }
+  ]
+}
+```
+
+For instructions to configure the domain access policy, see [Resource-based policies
+](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ac.html#ac-types-resource) in the Amazon OpenSearch Service documentation.
+
+### Fine-grained access control
+
+If your OpenSearch Service domain uses [fine-grained access control
+](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html), the OpenSearch sink plugin requires some additional configuration.
+
+#### IAM ARN as master user
```

Comment:
Global: Is "master user" the accepted term, or should it be something like "primary user"?

Reply:
Will switch to primary.

Reply:
Amending my previous comment. IAM still uses the term `master` when referring to the role of primary user (https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac-walkthrough-iam.html) Will hold off on switching this to remain consistent with the IAM offering.